components/ws-manager/pkg/manager/manager.go

```diff
@@ -808,8 +816,8 @@ func (m *Manager) StopWorkspace(ctx context.Context, req *api.StopWorkspaceReque
 
 // stopWorkspace stops a workspace. If the workspace is already stopping, this is a noop
 func (m *Manager) stopWorkspace(ctx context.Context, workspaceID string, gracePeriod time.Duration) (err error) {
-	if m.Config.DryRun {
-		log.WithFields(log.OWI("", "", workspaceID)).Info("should have stopped pod but this is a dry run")
+	if m.Config.MaintenanceMode {
```

Comment:
What does this mean for users? Can they not stop workspaces?

Reply:
Yes, they will not be able to stop until maintenanceMode is disabled, including timeouts that would have elapsed during the maintenance window.

Reply:
@aledbf let @jenting or I know if you need anything else? :pray:

Reply:
what happens to user's credits if they cannot stop workspace? :thinking:

Reply:
Good question @sagor999 . I believe we'll continue to burn through the credits when using this maintenance mode.

But, I think that's a side effect of not having mk2, and usage of this maintenance mode will be limited to Gitpod Dedicated for now. 

In other words, once we have mk2, I assume we'd persist the stop time request within the CR rather than return the failed precondition, and ideally leverage the time of the stop request to avoid burning through more credits.

Reply:
@aledbf @sagor999 Are there any other questions/concerns? If not, please help unhold it. Thank you.